install_sct

```diff
@@ -170,7 +170,7 @@ function check_requirements() {
       done
       if [[ $GCC_INSTALL =~ [Yy](es)? ]]; then
         if [[ ! $(which brew) ]]; then
-          yes | /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+          yes | /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
```

Comment:
hum, some users will don't understand this message:
~~~
julien-macbook:~ $ yes | /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
Need sudo access on macOS (e.g. the user julien to be an Administrator)!
~~~

Reply:
ahem... and it is failing when calling it with sudo (for another reason):
~~~
julien-macbook:~ $ sudo yes | /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
Password:
==> This script will install:
/usr/local/bin/brew
/usr/local/share/doc/homebrew
/usr/local/share/man/man1/brew.1
/usr/local/share/zsh/site-functions/_brew
/usr/local/etc/bash_completion.d/brew
/usr/local/Homebrew
==> Downloading and installing Homebrew...
remote: Enumerating objects: 3156, done.
remote: Counting objects: 100% (3156/3156), done.
remote: Compressing objects: 100% (21/21), done.
remote: Total 10136 (delta 3141), reused 3143 (delta 3135), pack-reused 6980
Receiving objects: 100% (10136/10136), 2.98 MiB | 7.98 MiB/s, done.
Resolving deltas: 100% (7815/7815), completed with 1254 local objects.
From https://github.com/Homebrew/brew
   e30ed4243..9db324ab7  master          -> origin/master
 * [new branch]          zsh-fpath-fixes -> origin/zsh-fpath-fixes
 * [new tag]             2.5.10          -> 2.5.10
 * [new tag]             2.5.11          -> 2.5.11
 * [new tag]             2.5.12          -> 2.5.12
 * [new tag]             2.5.6           -> 2.5.6
 * [new tag]             2.5.7           -> 2.5.7
 * [new tag]             2.5.8           -> 2.5.8
 * [new tag]             2.5.9           -> 2.5.9
 * [new tag]             2.6.0           -> 2.6.0
 * [new tag]             2.6.1           -> 2.6.1
 * [new tag]             2.6.2           -> 2.6.2
HEAD is now at 9db324ab7 Merge pull request #10045 from jonchang/remove-basic-autho
Error: 
  homebrew-core is a shallow clone.
  homebrew-cask is a shallow clone.
To `brew update`, first run:
  git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core fetch --unshallow
  git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask fetch --unshallow
This restriction has been made on GitHub's request because updating shallow
clones is an extremely expensive operation due to the tree layout and traffic of
Homebrew/homebrew-core and Homebrew/homebrew-cask. We don't do this for you
automatically to avoid repeatedly performing an expensive unshallow operation in
CI systems (which should instead be fixed to not use shallow clones). Sorry for
the inconvenience!
Failed during: /usr/local/bin/brew update --force
~~~

Reply:
well... this is a particular case because I _already_ had brew... maybe that wouldn't fail if brew was not installed already

Reply:
Just about that first part, that's because you'd need to run it as

```
yes | sudo /bin/bash -c ...
```
or

```
sudo sh -c "yes | bash -c ...."
```

Running `yes` as root doesn't grant brew's installer root.

Reply:
But the brew installer calls `sudo` inside of itself.

I think the other problem here is that `yes | ` is *not allowed*; that causes stdin to not be a tty (`isatty()`, or `tty -s`, fails), but `sudo` demands having a tty because it wants to prevent as much as possible people trying to save passwords to files and pasting them into it.

In #3102 I replaced `yes |` with using the noninteractive flags all the parts actually wanted, precisely because it's sketchy for this and similar cases.

So maybe #3102 does need to be merged first afterall.